src/states/common/base.rs

```diff
@@ -407,16 +407,14 @@ pub trait Base: Display {
     // Create HopMessage containing the given signed message.
     fn to_hop_message(&self, signed_msg: SignedRoutingMessage) -> Result<Message, RoutingError> {
         let hop_msg = HopMessage::new(signed_msg)?;
-        Ok(Message::Hop(hop_msg))
+        Ok(Message::Hop(Box::new(hop_msg)))
     }
 
     fn to_signed_direct_message(&self, content: DirectMessage) -> Result<Message, RoutingError> {
-        SignedDirectMessage::new(content, self.full_id())
-            .map(Message::Direct)
-            .map_err(|err| {
-                error!("{} - Failed to create SignedDirectMessage: {:?}", self, err);
-                err
-            })
+        Ok(Message::Direct(Box::new(SignedDirectMessage::new(
+            content,
+            self.full_id(),
```

Comment:
Why did we remove the log here?

Reply:
The log was for an error that cannot happen